test/abilities/mirror_armor.test.ts

```diff
@@ -0,0 +1,186 @@
+import { allAbilities } from "#app/data/ability";
+import { Abilities } from "#enums/abilities";
+import { BattlerIndex } from "#enums/battler-index";
+import { Moves } from "#enums/moves";
+import { Species } from "#enums/species";
+import { Stat, type BattleStat } from "#enums/stat";
+import { GameManager } from "#test/testUtils/gameManager";
+import Phaser from "phaser";
+import { afterEach, beforeAll, beforeEach, describe, expect, it, vi } from "vitest";
+
+describe("Abilities - Mirror Armor", () => {
+  let phaserGame: Phaser.Game;
+  let game: GameManager;
+
+  beforeAll(() => {
+    phaserGame = new Phaser.Game({
+      type: Phaser.HEADLESS,
+    });
+  });
+
+  afterEach(() => {
+    game.phaseInterceptor.restoreOg();
+  });
+
+  beforeEach(() => {
+    game = new GameManager(phaserGame);
+    game.override
+      .ability(Abilities.BALL_FETCH)
+      .battleType("single")
+      .disableCrits()
+      .enemySpecies(Species.MAGIKARP)
+      .enemyAbility(Abilities.MIRROR_ARMOR)
+      .enemyMoveset(Moves.SPLASH)
+      .startingLevel(100)
+      .enemyLevel(100);
+  });
+
+  it("should reflect moves' stat-lowering effects onto the source", async () => {
+    await game.classicMode.startBattle([Species.FEEBAS]);
+
+    const player = game.field.getPlayerPokemon();
+    const enemy = game.field.getEnemyPokemon();
+
+    game.move.use(Moves.GROWL);
+
+    await game.phaseInterceptor.to("BerryPhase", false);
+
+    expect(enemy.getStatStage(Stat.ATK)).toBe(0);
+    expect(player.getStatStage(Stat.ATK)).toBe(-1);
+  });
+
+  it("should reflect abilities' stat-lowering effects onto the source", async () => {
+    game.override.ability(Abilities.INTIMIDATE);
+
+    await game.classicMode.startBattle([Species.FEEBAS]);
+
+    const player = game.field.getPlayerPokemon();
+    const enemy = game.field.getEnemyPokemon();
+
+    expect(enemy.getStatStage(Stat.ATK)).toBe(0);
+    expect(player.getStatStage(Stat.ATK)).toBe(-1);
+  });
+
+  it("should not divert self-targeted stat-lowering effects", async () => {
+    await game.classicMode.startBattle([Species.FEEBAS]);
+
+    const enemy = game.field.getEnemyPokemon();
+
+    game.move.use(Moves.SPLASH);
+    await game.move.forceEnemyMove(Moves.OVERHEAT);
+    await game.setTurnOrder([BattlerIndex.ENEMY, BattlerIndex.PLAYER]);
+    await game.move.forceHit();
+
+    await game.phaseInterceptor.to("BerryPhase", false);
+    expect(enemy.getStatStage(Stat.SPATK)).toBe(-2);
```

Comment:
```suggestion
    expect(enemy.getStatStage(Stat.SPATK)).toBe(-2);
    expect(player.getStatStage(Stat.SPATK)).toBe(0);
```
for safety/consistency?